cli/pcluster/utils.py

```diff
@@ -754,3 +754,18 @@ def get_cluster_capacity(stack_name):
         if scheduler == "awsbatch"
         else get_asg_settings(stack_name).get("DesiredCapacity")
     )
+
+
+def get_efa_enabled_instance_types():
+    client = boto3.client("ec2")
+    instance_types = []
+    next_token = ""
+
+    while next_token is not None:
+        call_args = {"Filters": [{"Name": "network-info.efa-supported", "Values": ["true"]}], "NextToken": next_token}
+
+        response = client.describe_instance_types(**{k: v for k, v in call_args.items() if v})
```

Comment:
What about adding the arguments directly in the `describe_instance_types` call ?
You don't need to check if `next_token` is `None` because you're inside the `while` that is already checking for it.
Am I missing something? If so, could you please add a comment describing what are you doing within this line?

For the pagination (see @demartinofra suggestion) we have an utility function `paginate_boto3` in the `pcluster/utils.py` file.

Reply:
Thanks, switched to paginate_boto3.